src/main/java/ch/njol/skript/entity/FishData.java

```diff
@@ -53,7 +61,7 @@ public FishData(int pattern) {
 	protected boolean init(Literal<?>[] exprs, int matchedPattern, ParseResult parseResult) {
 		if (matchedPattern == 0)
 			wildcard = true;
-		pattern = (matchedPattern == 0) ? ThreadLocalRandom.current().nextInt(1, 5) : matchedPattern;
+		pattern = (matchedPattern == 0) ? ThreadLocalRandom.current().nextInt(1, HAS_TADPOLE ? 6 : 5) : matchedPattern;
```

Comment:
Since we have patterns list, can we use List#size() here?
```suggestion
		pattern = (matchedPattern == 0) ? ThreadLocalRandom.current().nextInt(1, patterns.size()) : matchedPattern;
```

Reply:
this suggestion specifically won't work, but ill rework the code so it will